posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -374,7 +408,51 @@ async def materialize_model(model_label: str, team: Team) -> tuple[str, DeltaTab
         prepare_s3_files_for_querying(saved_query.folder_path, saved_query.name, file_uris)
 
     key, delta_table = tables.popitem()
-    return (key, delta_table)
+
+    # Count rows and update both DataWarehouseTable and DataModelingJob
+    row_count = await asyncio.to_thread(count_delta_table_rows, delta_table)
+    await update_table_row_count(saved_query, row_count)
+
+    # Update the job record with the row count and completed status
+    job.rows_materialized = row_count
+    job.status = DataModelingJob.Status.COMPLETED
+    job.last_run_at = dt.datetime.now(dt.UTC)
+    await database_sync_to_async(job.save)()
```

Comment:
Doesn't look like we are capturing this data when one of the CH exceptions is raised. We probably want to at least not leave the status as Running?